File: frontend/entities/ActivityEvent.ts
import ParticipantEvent from "./ParticipantEvent";
import RaffleItem from "./RaffleItem";

export default class ActivityEvent {
  id: string;
  baseMinutes: number;
  eventName: string;
  minutesPerTicket: number;
  startTime: string;
  endTime: string;
  winnersVisible: boolean;
  prizesVisible: boolean;
  participantEvents: ParticipantEvent[];
  raffleItems: RaffleItem[];

  constructor(
    id?: string,
    baseMinutes?: number,
    eventName?: string,
    minutesPerTicket?: number,
    startTime?: string,
    endTime?: string,
    winnersVisible?: boolean,
    prizesVisible?: boolean,
    participantEvents?: ParticipantEvent[],
    raffleItems?: RaffleItem[]
  ) {
    if (
      id &&
      baseMinutes !== undefined &&
      eventName &&
      minutesPerTicket !== undefined &&
      startTime &&
      endTime &&
      winnersVisible !== undefined &&
      prizesVisible !== undefined &&
      participantEvents &&
      raffleItems
    ) {
      this.id = id;
      this.baseMinutes = baseMinutes;
      this.eventName = eventName;
      this.minutesPerTicket = minutesPerTicket;
      this.startTime = startTime;
      this.endTime = endTime;
      this.winnersVisible = winnersVisible;
      this.prizesVisible = prizesVisible;
      this.participantEvents = participantEvents;
      this.raffleItems = raffleItems;
    } else {
      this.id = "";
      this.baseMinutes = 0;
      this.eventName = "";
      this.minutesPerTicket = 0;
      this.startTime = "";
      this.endTime = "";
      this.winnersVisible = false;
      this.prizesVisible = false;
      this.participantEvents = [];
      this.raffleItems = [];
    }
  }
}

File: backend/App.Domain/ActivityEvent.cs
using App.Domain.Identity;
using Base.Domain;

namespace App.Domain;

public class ActivityEvent : BaseEntityId
{
    public Guid AppUserId { get; set; }
    public AppUser? AppUser { get; set; }
    public int BaseMinutes { get; set; }
    public string EventName { get; set; } = "";
    public int MinutesPerTicket { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime EndTime { get; set; }
    public bool WinnersVisible { get; set; }
    public bool PrizesVisible { get; set; }
    
    public ICollection<ParticipantEvent>? ParticipantEvents { get; set; }
    public ICollection<RaffleItem>? RaffleItems { get; set; }
}